### PR TITLE
Add Social Network accounts to your profile

### PR DESCRIPTION
**Pull request by timbl**
So that you can use the solid profile as your master profile

Work in Progess

### CODE REVIEW

**Comment by timbl** [comment]
Screenshot profile pane test 2024-11-05 at 09 05 30

**Comment by timbl** [comment]
basically  working but tests not yet.

**Comment by timbl** [comment]
I'm getting this bogus hanging "build (14.x) Expected — Waiting for status to be reported Required".  
Last time IIRC it was we had some rules on the repo, but here there don't seem to be any rules in the settings.

**Comment by timbl** [comment]
Should have mentioned earlier, this is in response to https://github.com/SolidOS/profile-pane/issues/51

File: src/editProfilePane/wrapped-profileFormText.ts [hunk 1]
```diff
@@ -383,6 +395,284 @@ WHERE
...
+:MediumAccount rdfs:subClassOf foaf:Account ;
+    rdfs:label "Medium";
+    foaf:userProfilePrefix "https://www.linkedin.com/";
+    foaf:icon <https://upload.wikimedia.org/wikipedia/commons/a/a5/Medium_icon.svg>;
+    foaf:homepage <https://linkedin.com/> .
+
+:NostrAccount rdfs:subClassOf foaf:Account ; ### FEDERATED
+    rdfs:label "Nostr";
+    foaf:userProfilePrefix "https://www.linkedin.com/";
+    foaf:icon <https://upload.wikimedia.org/wikipedia/commons/b/b9/2023_Facebook_icon.svg>;
```
Comment (melvincarvalho) [inline src/editProfilePane/wrapped-profileFormText.ts:636]: No icon in wiki yet, but we could add one.  Here is one:

https://raw.githubusercontent.com/mbarulli/nostr-logo/refs/heads/main/SVG/nostr-icon-purple-on-white.svg

File: src/editProfilePane/wrapped-profileFormText.ts [hunk 2]
```diff
@@ -383,6 +395,284 @@ WHERE
...
+    foaf:homepage <https://linkedin.com/> .
```
Comment (melvincarvalho) [inline src/editProfilePane/wrapped-profileFormText.ts:637]: There's no canonical homepage, but one I like is : https://nostr.net/

File: src/editProfilePane/wrapped-profileFormText.ts [hunk 3]
```diff
@@ -383,6 +395,284 @@ WHERE
...
+    rdfs:label "Pinterest";
+    foaf:userProfilePrefix "https://pinterest.com/";
+    foaf:icon <https://upload.wikimedia.org/wikipedia/commons/1/1e/The_reddit_2023_mini_logo.jpg>;
+    foaf:homepage <https://pinterest.com/> .
+
+:RedditAccount rdfs:subClassOf foaf:Account ;
+    rdfs:label "Reddit";
+    foaf:userProfilePrefix "https://www.reddit.com/user/";
+    foaf:icon <https://upload.wikimedia.org/wikipedia/commons/1/1e/The_reddit_2023_mini_logo.jpg>;
+    foaf:homepage <https://bsky.app/> .
```
Comment (melvincarvalho) [inline src/editProfilePane/wrapped-profileFormText.ts:649]: should be https://reddit.com/

**Comment by chunt007** [comment]
29:     foaf:Account a rdfs:Class;
30:          rdfs:label "Online Account Provider";
31:          owl:disjointUnionOf ( :BlueSkyAccount :FacebookAccount :GithubAccount :InstagramAccount
32:          :LinkedInAccount :MastodonAccount :MatrixAccount :PinterestAccount 
33:          :RedditAccount :SnapchatAccount :TiktokAccount  :TwitterAccount :OtherAccount) .

Don't forget Pinterest :P

File: src/editProfilePane/wrapped-profileFormText.ts (hunk 2 above)
Comment (timbl) [reply src/editProfilePane/wrapped-profileFormText.ts:637]: Hard to know how say how to connect with you on Nostr.  Give your key, .. which relays they follow?

File: src/editProfilePane/wrapped-profileFormText.ts (hunk 3 above)
Comment (timbl) [reply src/editProfilePane/wrapped-profileFormText.ts:649]: should be

File: src/editProfilePane/wrapped-profileFormText.ts [hunk 4]
```diff
@@ -383,6 +395,284 @@ WHERE
...
+    foaf:userProfilePrefix "https://matrix.to/";
+    foaf:homepage <https://matrix.org/> .
+
+:MediumAccount rdfs:subClassOf foaf:Account ;
+    rdfs:label "Medium";
+    foaf:userProfilePrefix "https://www.linkedin.com/";
+    foaf:icon <https://upload.wikimedia.org/wikipedia/commons/a/a5/Medium_icon.svg>;
+    foaf:homepage <https://linkedin.com/> .
+
+:NostrAccount rdfs:subClassOf foaf:Account ; ### FEDERATED
```
[the remark this answers is not in the document]
Comment (timbl) [reply src/editProfilePane/wrapped-profileFormText.ts]: Removed the word.

File: src/editProfilePane/wrapped-profileFormText.ts [hunk 5]
```diff
@@ -383,6 +395,284 @@ WHERE
...
+    rdfs:label "Nostr";
+    foaf:userProfilePrefix "https://www.linkedin.com/";
```
[the remark this answers is not in the document]
Comment (timbl) [reply src/editProfilePane/wrapped-profileFormText.ts]: Thanks. Implemented.

File: src/editProfilePane/wrapped-profileFormText.ts (hunk 1 above)
Comment (timbl) [reply src/editProfilePane/wrapped-profileFormText.ts]: Took a copy of that for the solid-ui icon library.

**Comment by chunt007** [comment]
Not sure if this was brought up, but for some reason, the profile picture does not seem to work when you log into a solid community account. I merged the pull requests into my local repo and can't see the profile picture anymore.

**Comment by timbl** [comment]
> Not sure if this was brought up, but for some reason, the profile picture does not seem to work when you log into a solid community account. I merged the pull requests into my local repo and can't see the profile picture anymore.

In what context? In the `npm start` or in a version of SolidOS which has it running?  Where is the code running,and which was the profile being looked at?  Could it be a permissions problem I wonder ..

**Comment by chunt007** [comment]
> > Not sure if this was brought up, but for some reason, the profile picture does not seem to work when you log into a solid community account. I merged the pull requests into my local repo and can't see the profile picture anymore.
> 
> In what context? In the `npm start` or in a version of SolidOS which has it running? Where is the code running,and which was the profile being looked at? Could it be a permissions problem I wonder ..

developerconsole

Update: Can't get updatability status <https://chunt.solidcommunity.net/IndividualChats/chunt.solidcommunity.net/index.ttl> before patching: Error: Fetcher: <https://chunt.solidcommunity.net/IndividualChats/chunt.solidcommunity.net/index.ttl> Unauthenticated
    at http://localhost:8080/main.js:51724:21

**Comment by timbl** [comment]
@chunt007 Yes in general i f you are not logged in, then you can't load your preferences file (or anyone else's).  Solidos will try to load it as it gives it more info, but it should fail silently.  You'll see fetch errors in the console log, but should not on the UI

**Comment by timbl** [comment]
Upgraded dependency on solid-ui to 2.5 to make sure we have the new icons

**Comment by timbl** [comment]
I think it is time to merge this one, to add the social net link functionality

**Pull request merged**